pkg/client/harbor/harbor_test.go

```diff
@@ -0,0 +1,217 @@
+package harbor_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/bitnami-labs/charts-syncer/api"
+	"github.com/bitnami-labs/charts-syncer/internal/cache"
+	"github.com/bitnami-labs/charts-syncer/internal/chartrepotest"
+	"github.com/bitnami-labs/charts-syncer/internal/utils"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/harbor"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/types"
+	"helm.sh/helm/v3/pkg/time"
+)
+
+var (
+	harborRepo = &api.Repo{
+		Kind: api.Kind_HARBOR,
+		Auth: &api.Auth{
+			Username: "user",
+			Password: "password",
+		},
+	}
+)
+
+func prepareTest(t *testing.T) (*harbor.Repo, error) {
```

Comment:
you should consider using [testing.M](https://golang.org/pkg/testing/#hdr-Main). See [this article](https://www.ardanlabs.com/blog/2019/10/integration-testing-in-go-set-up-and-writing-tests.html#creating-testmain-using-testing-m) for a complete example.

Reply:
ok, I will check it after merging this PR and create a new one applying the changes to all the affected clients if it applies nicely